modP2pImpl/src/org/aion/p2p/impl1/tasks/TaskInbound.java

```diff
@@ -155,12 +153,11 @@ public void run() {
         p2pLOG.info("p2p-pi shutdown");
     }
 
-    private void accept() throws Exception {
+    private void accept(SelectableChannel _channel) throws Exception {
         if (this.nodeMgr.activeNodesSize() >= this.mgr.getMaxActiveNodes()) {
             return;
         }
-
-        SocketChannel channel = this.tcpServer.accept();
+        SocketChannel channel = ((ServerSocketChannel) _channel).accept();
```

Comment:
I feel like there should be an instance-of check before doing the casting. Perhaps the method should receive a `ServerSocketChannel` parameter and the calling method above can do the check and casting.